control/conversion_test.go

```diff
@@ -24,81 +28,27 @@ func TestJSONToProto(t *testing.T) {
 		Token:  Token("oqiwdJKNsdKIUwezd92DNQsndkDERDFKJNQWSwq3rODIU"),
 		Sender: sender, Recipients: recipients, Cc: cc, Bcc: bcc,
 		MinPeriod: 0.0001, MaxSize: 10000000}
-	converted := JSONToProto(jsonChan)
+	converted := JSONToProto(&jsonChan)
 
-	if jsonChan.Descriptor != Descriptor(converted.Descriptor_) {
-		t.Errorf("expected descriptor to be %v, got %v",
-			jsonChan.Descriptor, converted.Descriptor_)
-	}
-	if jsonChan.Token != Token(converted.Token) {
-		t.Errorf("expected token to be %v, got %v",
-			jsonChan.Token, converted.Token)
+	bts, err := json.Marshal(jsonChan)
+	if err != nil {
+		t.Fatalf("failed to marhsal the channel: %s", err.Error())
 	}
+	jsonChanStr := string(bts)
 
-	if *jsonChan.Sender.Name != converted.Sender.Name {
-		t.Errorf("expected sender name to be %v, got %v",
-			*jsonChan.Sender.Name, converted.Sender.Name)
-	}
-	if jsonChan.Sender.Email != converted.Sender.Email {
-		t.Errorf("expected sender email to be %v, got %v",
-			jsonChan.Sender.Email, converted.Sender.Email)
+	marshaler := jsonpb.Marshaler{OrigName: false}
+	protoChanStr, err := marshaler.MarshalToString(converted)
+	if err != nil {
+		t.Fatalf("failed to marhsal the protocol buffer: %s", err.Error())
 	}
 
-	if len(jsonChan.Recipients) != len(converted.Recipients) {
-		t.Errorf("expected sender to be %v, got %v",
-			jsonChan.Recipients, converted.Recipients)
-	} else {
-		for i, recipient := range jsonChan.Recipients {
-			if recipient.Email != converted.Recipients[i].Email {
-				t.Errorf("expected recipient %d email to be %v, got %v",
-					i, recipient.Email, converted.Recipients[i].Email)
-			}
-			if *recipient.Name != converted.Recipients[i].Name {
-				t.Errorf("expected recipient %d name to be %v, got %v",
-					i, *recipient.Name, converted.Recipients[i].Name)
-			}
-		}
-	}
+	protoChanStr = strings.Replace(
```

Comment:
Unfortunately I had to change this manually. The protobuf library does not convert automatically variable names from camel to snake case.

Reply:
This is dangerous in case there is ever "minPeriod" in the value a field _etc._ What is more, the logic of this test prevents you from having some fields in the database stored as a protobuf and remove them in the conversion.

I would suggest to decouple these and just have an expected string what a JSON marshalling should look like, and then also what a protobuf marshalling should look like and compare against these. In other words, you would codify the expected outputs of the conversion functions as strings and validate against these golden strings.

Whenever the structure would change, the "golden" strings would need to change as well, but I don't expect that to be a lot of work.